source/neuropod/bindings/java/src/main/java/com/uber/neuropod/Dimension.java

```diff
@@ -65,4 +67,30 @@ public long getValue() {
     public String getSymbol() {
         return symbol;
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (!(o instanceof Dimension)) return false;
+        Dimension dimension = (Dimension) o;
+        return getValue() == dimension.getValue() &&
+                Objects.equals(getSymbol(), dimension.getSymbol());
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(getValue(), getSymbol());
+    }
+
+    @Override
+    public String toString() {
+        if (value >= 0) {
+            return String.valueOf(value);
+        } else if (value == -1) {
+            return "None";
+        } else if (value == -2) {
+            return symbol;
+        }
+        return "Not Set!";
```

Comment:
Should we throw an error here? This case means the `value < -2` which likely means something went wrong